src/Storages/MergeTree/MergeTreeSettings.h

```diff
@@ -130,6 +130,7 @@ struct Settings;
     M(UInt64, vertical_merge_algorithm_min_columns_to_activate, 11, "Minimal amount of non-PK columns to activate Vertical merge algorithm.", 0) \
     \
     /** Compatibility settings */ \
+    M(Bool, allow_suspicious_indices, false, "Reject primary/secondary indexes and sorting keys with identical expressions", 0) \
```

Comment:
What's the point of having the same settings in `Settings` and `MergeTreeSettings`

Reply:
I remember the problem was that
- `CREATE TABLE ... SETTINGS ...` only accepts mergetree settings and
- `ALTER TABLE ... SETTINGS ...` only accepts general settings.

To keep things consistent for users, `allow_suspicious_indices` was added to both.

In retrospect, I agree it can be confusing. Either we can
- keep it this way \+ mention in the setting's description that the other setting should be enabled as well
- keep only one kind of setting, ideally the general one. One then needs to use CREATE TABLE like this: `SET allow_suspicious_indices = 1; CREATE TABLE ...;` but that would acceptable IMO.

Reply:
> CREATE TABLE ... SETTINGS ... only accepts mergetree settings and

No problem at all:
```
SET whatever_general_setting=1;
CREATE TABLE no_more_settings ...
```

We cannot simply remove a setting because someone might enable it in the config, so their server will fail to start after upgrading due to "unknown setting" exception.

Let's keep both, but respect the setting from the global context too, and remove the check when applying committed changes

Reply:
> No problem at all:

Another option:
```
CREATE TABLE weird_syntax ... SETTINGS index_granularity=8192 SETTINGS general_setting=42;
```